scaleway/data_source_instance_servers.go

```diff
@@ -0,0 +1,74 @@
+package scaleway
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func dataSourceScalewayInstanceServers() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceScalewayInstanceServersRead,
+		Schema: map[string]*schema.Schema{
+			"tags": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "Tags list servers with these exact tags (to filter with several tags, use commas to separate them)",
+			},
+			"instances": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+						"public_ip": {
+							Computed: true,
+							Type:     schema.TypeString,
+						},
+					},
+				},
+			},
+			"zone":            zoneSchema(),
+			"organization_id": organizationIDSchema(),
+			"project_id":      projectIDSchema(),
+		},
+	}
+}
+
+func dataSourceScalewayInstanceServersRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	instanceAPI, zone, err := instanceAPIWithZone(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	res, err := instanceAPI.ListServers(&instance.ListServersRequest{
+		Zone:    zone,
+		Name:    expandStringPtr(d.Get("name")),
```

Comment:
Can you add some test about:

1. With name
2. Without name
3. With tags `[]`
4. With other zone